### PR TITLE
Update Mpoly printing

### PR DESCRIPTION
**Pull request by lgoettgens**
Currently not tested. I'll mark as ready to review once tested against Oscar locally.

### CODE REVIEW

File: src/MPoly.jl [hunk 1]
```diff
@@ -560,6 +560,8 @@ end
 @enable_all_show_via_expressify MPolyRingElem
 
 function show(io::IO, ::MIME"text/plain", p::MPolyRing)
+  @show_name(io, W)
+  @show_special(io, W)
```
Comment (joschmitt) [inline src/MPoly.jl:564]: `@show_special(io, MIME"text/plain"(), W)`

**Comment by lgoettgens** [comment]
Yes, indeed, this fixes the behaviour observed in https://github.com/oscar-system/Oscar.jl/pull/3288#discussion_r1473510149. If we release this before merging https://github.com/oscar-system/Oscar.jl/pull/3288, this won't break anything.

**Review by joschmitt — APPROVED** [review]
Looks good to me. Can't say I fully understand this `@show_...` magic though.

**Comment by thofma** [comment]
No clue about this. If Johannes is happy, I am happy.

**Pull request merged**